vignettes/MSnbase-centroiding.Rmd

```diff
@@ -0,0 +1,333 @@
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r style, echo = FALSE, results = 'asis'}
+library(BiocStyle)
+BiocStyle::markdown()
+```
+
+# Introduction
+
+Profile raw data detected by a mass spectromer consists of a distributed signal
+across m/z values at each point where an ion is detected (which depends on the
+resolution and settings of the instrument) [@Smith:2014di]. Profile mode data
+can be processed into *centroid* data by retaining only the local maximum in
+each range in which an ion is detected. This *centroiding* largely reduces the
+amount of data without much loss in information content. Certain algorithms,
+such as the *centWave* method in the `r Biocpkg("xcms")` package for
+chromatographic peak detection in LC-MS experiments, require the data to be in
+centroid mode.
+ 
+For mass spectrometry systems from many manufacturers centroiding can already be
+performed within the vendor's software during or after data acquisition. The
+software from some vendors does however generate centroided data of poor
+quality. `MSnbase` provides some base functionality to perform centroiding of
+profile MS data, and, with the possibility of exporting the centroided MS data
+to *mzML* files, the results of this centroiding can also be used as input for
+other software, if needed.
+
+# Centroiding of profile-mode MS data
+
+In this vignette we use a subset of a profile-mode LC-MS data of pooled human
+serum samples measured on a AB Sciex TripleTOF 5600+ mass spectrometer (the
+employed chromatography was a hydrophilic interaction high-performance liquid
+chromatography (HILIC HPLC)). The mzML file contains profile mode data for an
+m/z range from 105 to 130 and a retention time from 0 to 240 seconds. For more
+details on the sample see `?msdata::sciexdata`. Below we load the required
+libraries and read the MS data.
+
+```{r libraries, message = FALSE}
+library(MSnbase)
+library(msdata)
+
+fl <- dir(system.file("sciex", package = "msdata"), full.names = TRUE)[2]
+data_prof <- readMSData(fl, mode = "onDisk", centroided = FALSE)
+```
+
+We next extract the profile MS data for the [M+H]+ adduct of serine with the
+expected m/z of 106.049871. We thus filter the `data_prof` object using an m/z
+range containing the signal for the metabolite and a retention time window from
+175 to 187 seconds corresponding to the time when the analyte elutes from the
+LC. Subsequently we plot the profile MS data for serine.
+
+```{r serine-msdata, fig.align = "center", fig.cap = "MS profile data for serine. Upper panel shows the base peak chromatogram (BPC), lower panel the individual signals in the retention time - m/z space. The horizontal dashed red line indicates the theoretical m/z of the [M+H]+ adduct of serine."}
+## Define the mz and retention time ranges
+serine_mz <- 106.049871
+mzr <- c(serine_mz - 0.01, serine_mz + 0.01)
+rtr <- c(175, 187)
+
+## Filtering the object
+serine <- filterMz(filterRt(data_prof, rt = rtr), mz = mzr)
```

Comment:
I would suggest to use piping here:
```r
serine <- data_prof %>%
	filterRt(rtr) %>%
	filterMz(mzr)
```

Reply:
Yes, that's easier to read. Will change.

Reply:
But note that we'll then have to add `magrittr` also to `SUGGESTS`

Reply:
Yes, that's OK.

Reply:
All done in https://github.com/lgatto/MSnbase/pull/315/commits/6935bf98eda44bfdcbd7a853065bf29236772e97